docs_src/kafka/custom_encoding/app.py

```diff
@@ -17,19 +14,13 @@ class Data(BaseModel):
     )
 
 
-async def custom_parser(
-    msg: ConsumerRecord, original: Callable[[ConsumerRecord], KafkaMessage]
-) -> KafkaMessage:
-    kafka_msg = await original(msg)
-
+async def custom_decoder(msg: KafkaMessage) -> Data:
     if hasattr(Data, "model_validate_json"):
-        kafka_msg.body = Data.model_validate_json(kafka_msg.body.decode("utf-8"))
+        return Data.model_validate_json(msg.body.decode("utf-8"))
     else:
-        kafka_msg.body = Data.parse_raw(kafka_msg.body.decode("utf-8"))
-
-    return kafka_msg
+        return Data.parse_raw(msg.body.decode("utf-8"))
 
 
-@broker.subscriber("input_data", parser=custom_parser)
+@broker.subscriber("input_data", decoder=custom_decoder)
```

Comment:
Parser transorms ConsumerRecord -> StreamMessage, but the message body is serialized by decoder